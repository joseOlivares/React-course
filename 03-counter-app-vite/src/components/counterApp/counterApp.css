.customButton {
    color: #61dafb;
    background-color: transparent;
    border: 2px solid #61dafb;  
    padding: 10px 20px;
    border-radius: 10px;
    cursor: pointer; 
}

.m-l {
    margin-left: 10px;
}