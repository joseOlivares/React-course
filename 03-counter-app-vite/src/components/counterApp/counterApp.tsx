
import './counterApp.css'; 
import { useState} from 'react';  //hook useState


interface MyProp {
    value: number
}

export const CounterApp = ({value}: MyProp) => {
    const [counter, setCounter] = useState(value); //Aqui se maneja el estado del componente

    const addOne=()=>{
        setCounter(counter+1);
    }
    const substractOne=()=>{
        setCounter((c)=>c-1); //esta es otra forma
    }


    return (
        <>
            <h1>CounterApp</h1>
            <h2> {counter}</h2>  

            <button className="customButtom" onClick={addOne}>+1</button>
            <button className="customButtom m-l" onClick={substractOne}>-1</button>
            <button className="customButtom m-l" onClick={()=>setCounter(value)}>Reset</button>

        
        </>

    )
}   




