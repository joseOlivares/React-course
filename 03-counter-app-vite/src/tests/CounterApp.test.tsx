import { screen, render, fireEvent } from "@testing-library/react";
import { CounterApp } from "../components/counterApp/counterApp";

describe('Testing CounterApp', () => {



    test('debe de hacer match con el snapshot', () => {
        const value = 10;
        const {container} = render(<CounterApp value={value}/>);
        expect(container).toMatchSnapshot();
    })

    test('debe de mostrar el valor inicial de 100', () => {
        const value = 100;
      render(<CounterApp value={value}/>);
        expect(screen.getByText(value)).toBeTruthy();
    })

    test('debe de incrementar con el boton +1', () => {
        const value = 10;
        render(<CounterApp value={value}/>);
        fireEvent.click(screen.getByText('+1'));
        expect(screen.getByText('11')).toBeTruthy();
    })
    
})