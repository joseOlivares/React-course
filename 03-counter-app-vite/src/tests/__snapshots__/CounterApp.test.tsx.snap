// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Testing CounterApp debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    10
  </h2>
  <button
    class="customButtom"
  >
    +1
  </button>
  <button
    class="customButtom m-l"
  >
    -1
  </button>
  <button
    class="customButtom m-l"
  >
    Reset
  </button>
</div>
`;
